### ISSUE TITLE
Critical Security Patch url-parse 1.5.10

### ISSUE THREAD

**Issue by suhail-n** [amqp-node/amqplib#674]
## Problem

The current url-parse version 1.5.3 has the following critical vulnerabilities [CVE-2022-0691](https://nvd.nist.gov/vuln/detail/CVE-2022-0691), and [CVE-2022-0686](https://nvd.nist.gov/vuln/detail/CVE-2022-0686)

## Fix

This problem has been resolved in url-parse@1.5.10

Change package.json from
```
 "dependencies": {
    "bitsyntax": "~0.1.0",
    "bluebird": "^3.7.2",
    "buffer-more-ints": "~1.0.0",
    "readable-stream": "1.x >=1.1.9",
    "url-parse": "~1.5.3"
  },
```
to:
```
 "dependencies": {
    "bitsyntax": "~0.1.0",
    "bluebird": "^3.7.2",
    "buffer-more-ints": "~1.0.0",
    "readable-stream": "1.x >=1.1.9",
    "url-parse": "~1.5.10"
  },
```

**Comment by tbreeding** [comment]
I updated it locally and ran the tests and they all passed.

**Comment by michal-ayash** [comment]
We're blocked on that issue. Do you have ETA?

**Comment by suhail-n** [comment]
> We're blocked on that issue. Do you have ETA?

On your project try the command `npm update url-parse`

This shouldn't create any breaking changes. I applied this fix for my project and now I'm waiting for the fix to be applied to amqplib.

**Comment by davidsousa5** [comment]
> > We're blocked on that issue. Do you have ETA?
> 
> On your project try the command `npm update url-parse`
> 
> This shouldn't create any breaking changes. I applied this fix for my project and now I'm waiting for the fix to be applied to amqplib.

Do you have any idea when do you will release a new version with a fix for this issue?

**Comment by michal-ayash** [comment]
Any updae??

**Comment by kibertoad** [comment]
[This is](https://github.com/amqp-node/amqplib/pull/671) PR is blocked by CI failing. Anyone who could contribute fix for that would be very welcome.

**Comment by suhail-n** [comment]
If anyone needs the fix right away, I published a fork with the fix on [NPM](https://www.npmjs.com/package/@suhailn/amqplib)

### Installation
`npm i @suhailn/amqplib`

### Usage
```javascript
const amqplib = require('@suhailn/amqplib')
```

### Pull Request

I also went and published my pull request in [PR-675](https://github.com/amqp-node/amqplib/pull/675)

**Comment by suhail-n** [comment]
The change has been merged in. See [PR-675](https://github.com/amqp-node/amqplib/pull/675)

**Comment by michal-ayash** [comment]
@suhail-n, we need the fix in the original branch. any ETA?

**Comment by michal-ayash** [comment]
@suhail-n as we need the formal npm package and not this fork - I opened a new issue [here](https://github.com/amqp-node/amqplib/issues/676), please handle that and don't close until you fix this in the original branch

**Comment by suhail-n** [comment]
The change should be in main branch. My merge was successful. That's why I closed it.
